sdk/nodejs/resource.ts

```diff
@@ -149,6 +186,28 @@ export abstract class Resource {
                 opts.protect = opts.parent.__protect;
             }
 
+            for (const parentAlias of opts.parent.__aliases) {
+                if (!opts.aliases) {
+                    opts.aliases = [];
```

Comment:
very much *not* a fan of mutating this data in place.  can we not just extract to a local and do all our work there?

Reply:
Addressed this by cloning the `opts` before modifying.